### PR TITLE
Remove SUIL dependency if no GUI framework is enabled and SUIL is not available

### PR DESCRIPTION
**Pull request by twischer-adit**

### CODE REVIEW

File: src/jalv.c [hunk 1]
```diff
@@ -1006,6 +1018,7 @@ main(int argc, char** argv)
 			jalv.world, &jalv.map, lilv_plugin_get_uri(jalv.plugin));
 	}
 
+#ifdef HAVE_SUIL
```
[the remark this answers is not in the document]
Comment (twischer-adit) [reply src/jalv.c]: `suil_ui_supported` will be defined by SUIL. Therefore I have to keep the ifdefs. But I will move them to allow the else if clause

File: src/jack.c [hunk 2]
```diff
@@ -223,10 +223,12 @@ jack_process_cb(jack_nframes_t nframes, void* data)
 					jack_midi_event_write(buf, frames, body, size);
 				}
 
+#ifdef HAVE_SUIL
```
[the remark this answers is not in the document]
Comment (twischer-adit) [reply src/jack.c]: I will change it

File: src/jalv.c [hunk 3]
```diff
@@ -434,15 +436,18 @@ jalv_set_control(const ControlID* control,
 		lv2_atom_forge_atom(&forge, size, type);
 		lv2_atom_forge_write(&forge, body, size);
 
+#ifdef HAVE_SUIL
```
[the remark this answers is not in the document]
Comment (twischer-adit) [reply src/jalv.c]: I will change it. But is a little bigger change due to the first parameter type is SUIL specific.

File: src/jalv.c [hunk 4]
```diff
@@ -655,12 +660,15 @@ jalv_send_to_ui(Jalv*       jalv,
 		return false;
 	}
 }
+#endif
 
 bool
 jalv_run(Jalv* jalv, uint32_t nframes)
 {
+#ifdef HAVE_SUIL
```
[the remark this answers is not in the document]
Comment (twischer-adit) [reply src/jalv.c]: I will change this

File: src/state.c [hunk 5]
```diff
@@ -175,9 +175,12 @@ set_port_value(const char* port_symbol,
 		return;
 	}
 
+#ifdef HAVE_SUIL
```
[the remark this answers is not in the document]
Comment (twischer-adit) [reply src/state.c]: Was required due to the SUIL parameter type of `jalv_ui_write()`. I will change it

**Comment by twischer-adit** [comment]
Mainly I have changed the type of the first parameter of `jalv_ui_write()`. Therefore I was able to shrink the ifdef parts.
What do you think about it?

**Comment by drobilla** [comment]
That looks better, though defining functions in the `suil` "namespace" in Jalv is odd.  Something like this should be Good Enough(TM) though, I'll tinker with it and try to get something merged this week.

**Comment by twischer-adit** [comment]
@drobilla 
> That looks better, though defining functions in the suil "namespace" in Jalv is odd. Something like this should be Good Enough(TM) though, I'll tinker with it and try to get something merged this week.

Okay. Thanks. Give me a ping if I should do anything.

**Comment by drobilla** [comment]
That's the cleanest I could get it.  Seems to work as far as my limited testing can tell... thoughts?

**Comment by twischer-adit** [comment]
Yes it is also fine for me. It is compiling and running on Linux ARM64 with JACK

**Pull request merged**